### ISSUE TITLE
Bad Fast5 - Half of total reads

### ISSUE THREAD

**Issue by vikcrypto** [jts/nanopolish#807]
Hi,

I am trying to run nanopolish on my dataset. Everything seem to be going fine until the consensus step. The nanopolish indexing seem to be fine. I get all the expected files after indexing with the message:

**$ nanopolish index -d Post_Run_Fast5/ -s sequencing_summary.txt XL280_Post_run.fastq
[readdb] indexing Post_Run_Fast5/
[readdb] num reads: 1611286, num reads with path to fast5: 1611286.**

It generates 4 files:
XL280_Post_run.fastq.index
XL280_Post_run.fastq.index.fai
XL280_Post_run.fastq.index.gzi
XL280_Post_run.fastq.index.readdb

The alignment works fine and I get a bam file - I have tried both bwa and minimap2 for this step but it did not seem to make any difference. The bam file is not empty - I made sure of that.

Then I ran the following command for consensus calling:
**$ nanopolish_makerange.py XL280_10.contigs.fasta | parallel --results nanopolish.results -P 8 nanopolish variants --consensus -o polished.{1}.vcf -w {1} -r XL280_Post_run.fastq -b XL280.bwa-sorted.bam -g XL280_10.contigs.fasta -t 4**

Thats when I run into issues and I get the output like this - 

**[post-run summary] total reads: 5580, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 2790
[post-run summary] total reads: 5748, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 2874
[post-run summary] total reads: 5794, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 2897
[post-run summary] total reads: 6100, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 3050
[post-run summary] total reads: 6122, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 3061
[post-run summary] total reads: 5602, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 2801**

Exactly half of the reads seem to be bad. Despite that, I get .vcf files - each of which is just 1 kb and does not have any information.
I looked into the nanopolish results folder and open individual folders. I find three files, two of which seem to have some errors that look like this:

seq file - 
$ /path/to/folder/nanopolish.results/1/tig00000003\:0-50200/seq ; exit;
**/path/to/folder/nanopolish.results/1/tig00000003:0-50200/seq: line 1: 1: command not found**
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
[Process completed]

stderr file:
$ /path/to/folder/nanopolish.results/1/tig00000003\:0-50200/stderr ; exit;
**/path/to/folder/nanopolish.results/1/tig00000003:0-50200/stderr: line 1: [post-run: command not found**
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
[Process completed]

I am not sure what is wrong. Any suggestions? Thank you.

**Comment by jts** [comment]
That's odd, can you paste a few lines from the `XL280_Post_run.fastq.index.readdb` file?

Jared

**Comment by vikcrypto** [comment]
Hi Jared,

Thanks for replying. Here are some lines from readdb file:
000003cc-8535-4714-ba59-4449543f10b5	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_327.fast5
00002044-c846-4095-9074-a7a97eb98cfd	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_371.fast5
0000222a-72a6-4721-9bf1-0d26c696ba56	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_289.fast5
000025e7-a43d-4d8b-a44b-e4ca4ae79fb7	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_269.fast5
00002c1c-bfc0-4bfd-ac80-83c6cbffb75f	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_405.fast5
00003627-ff0a-457e-9947-bc3aa5a17246	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_372.fast5
00003f5f-71e4-4b71-b59c-a8c4ab41e121	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_217.fast5
000046be-7ae4-456d-82e6-107d673dc082	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_180.fast5
00004cfd-a8e1-420f-904b-3c73cb5f658d	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_85.fast5
00004e98-cc3a-43f0-826e-3697eeace1e9	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_434.fast5
00005168-885a-4a43-8a39-b7c5897877f8	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_248.fast5
000058d5-8d36-4f9f-a6fa-e83c688e3ce5	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_239.fast5
00005efc-3487-47f8-85e9-fa30489d1238	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_142.fast5
0000617f-f22d-4603-9e09-47cae66602a0	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_144.fast5
00006a1c-ca0f-4493-a558-8a7dec5ce6cd	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_25.fast5
0000743a-ba4a-4fc7-aee1-e6a8beb7b0ac	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_354.fast5
00007d8f-4046-41da-8dcd-db357e4af878	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_31.fast5
00009273-c34b-437b-99a3-b1cd60dd9f0b	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_308.fast5
0000969d-ec4f-4adc-9f3d-204c64e210b5	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_224.fast5
00009761-974e-4364-b5dc-b9d9babe2d60	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_224.fast5
00009871-0358-42ad-bf87-012df02c5ef0	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_107.fast5
0000ad1f-1fcd-487b-9fb8-6293d9451100	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_415.fast5
0000bc76-e47d-46a0-8f04-610985f6d4af	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_61.fast5
0000c8f2-4f61-484a-ba57-2d2e72d958c9	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_362.fast5
0000cb69-ab2a-4b9e-8a74-482df82ec779	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_195.fast5
0000e0c8-dc9c-4ed3-b11d-d4f0cbd13146	Post_Run_Fast5//recovered_6b80c12e-c340-44d1-8ef8-e30a520a8f3f_413.fast5

Vikas

**Comment by Damien-hash** [comment]
Hi,

I am experiencing the same type of issue did you happened to find a solution ?

nanopolish variants --min-flanking-sequence 10 -x 1000000 --progress -t 1 --reads /home/genomique-avl12/Sortie_Resultats/TEST/articNcovNanopore_sequenceAnalysisNanopolish_articGuppyPlex/20210916-COVIDSEQ174_barcode03.fastq -o 20210916-COVIDSEQ174.nCoV-2019_2.vcf -b 20210916-COVIDSEQ174.trimmed.rg.sorted.bam -g /home/genomique-avl12/Sortie_Resultats/Test2/nCoV-2019/V3/nCoV-2019.reference.fasta -w "MN908947.3:1-29904" --ploidy 1 -m 0.15 --read-group nCoV-2019_2

[post-run summary] total reads: 15540, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 7770

**Comment by jts** [comment]
Hi,

Please see here: https://github.com/jts/nanopolish/issues/908#issuecomment-865078582

Jared

**Comment by rickyleungyk** [comment]
Hi Jared,

I run into the same problem. The problem you cited in #908 was kind of different from mine. I got 
[readdb] num reads: 900132, num reads with path to fast5: 900132
but I still got exactly half of the fast5 bad after I ran eventalign:
[post-run summary] total reads: 395136, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 197568

I double check all my past fast5 and fastq files. No duplication was found. I have not been able to find any ways to fix this issue. Do you have any suggestions?

Thanks,
Ricky

**Comment by ric-89** [comment]
Hello,
I am experiencing the same problem when I try to run nanopolish call-methylation.
I map all of my 159,952 reads with winnowmap. Then I select only the reads mapping on a specific genomic region and save them in a separate REGION.fastq file (with 44,194 reads) as well as in a separate REGION.sorted.bam file.

Then I run 
nanopolish index -v -d $FAST5DIRECTORY REGION.fastq
After indexing:
[readdb] num reads: 159,952, num reads with path to fast5: 159,952
The .readdb file contains 159,952 lines, which corresponds to the total amount of reads in my dataset. Interestingly, all reads are indexed, not only the ones I had subselected in the REGION.fastq file. In any case, it seems all reads can find the corresponding fast5 information.


Then I run:
nanopolish call-methylation -r REGION.fastq -b REGION.sorted.bam -g $REFDIR/$REFNAME > output.methylation_calls.tsv 
And I get this:
[post-run summary] total reads: 33320, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 16660

The "bad fast5" is exactly half of the "total reads" number. The output output.methylation_calls.tsv file does not have any methylation calls, it is just the header without any other lines.
In addition, I struggle to understand why 33320 reads are taken into account, when my input fastq file has 44194 reads. 

I doublechecked that my fastq or fast5 are not duplicated. Maybe I am misunderstanding something of how this tool work.
If anybody has found any solution that would be really helpful.
Thanks
Ric
